cli/tests/unit/serve_test.ts

```diff
@@ -2887,38 +2887,48 @@ Deno.test(
   },
 );
 
-// TODO(fbaltor): As it is this test should throw and fail
 // https://fetch.spec.whatwg.org/#dom-request-clone
 Deno.test({
   name: "Throw if locked",
-  ignore: true,
   permissions: { net: true },
   fn: async function shouldThrowIfBodyIsUnusableLocked() {
     const ac = new AbortController();
     const listeningPromise = deferred();
 
     const server = Deno.serve({
-      handler: (req) => {
+      handler: async (req) => {
         const _reader = req.body?.getReader();
 
-        req.clone();
+        try {
+          req.clone();
+        } catch (error) {
+          assert(error instanceof TypeError);
+          assert(
+            error.message.endsWith("Body is unusable."),
+          );
 
+          ac.abort();
+          await server.finished;
+        }
         return new Response("ok");
       },
       signal: ac.signal,
       onListen: ({ port }: { port: number }) => listeningPromise.resolve(port),
     });
 
-    const port = await listeningPromise;
-    const resp = await fetch(`http://localhost:${port}/`, {
-      headers: { connection: "close" },
-      method: "POST",
-      body: '{"bar":true}',
-    });
-
-    ac.abort();
-    await resp.body?.cancel();
-    await server.finished;
+    try {
+      const port = await listeningPromise;
+      await fetch(`http://localhost:${port}/`, {
+        headers: { connection: "close" },
+        method: "POST",
+        body: '{"bar":true}',
+      });
+    } catch (e) {
```

Comment:
Thanks for the follow-up. A few notes/questions:

What gets caught here? 

Could you add an error message assertion here (same as you have in the serve block), and `assert(false)` at the end of the try block to ensure that the exception is always thrown?

Reply:
This handler is catching the "client" error when [the connection is closed by the server during it's catch](https://github.com/fbaltor/deno/blob/d99adc9def150a01af7487ab2b66b6d18dc037dd/cli/tests/unit/serve_test.ts#L2910-L2911). It throws the following:

![image](https://github.com/denoland/deno/assets/9384697/41be1d6b-2d53-4b0a-875c-12e995976632)

It is kind of a side effect.

Reply:
I was thinking that maybe I could just catch this "silently". Since it is not the focus of the test sounds a mix of concerns to me, but I'm not sure.

Reply:
I think that's reasonable, but we should add a `fail` and a note to help future readers of the code understand this. I think we should actually be returning a 500 from this particular case, so I think you may have discovered a subtle bug if the `Deno.serve` API is misused.

Reply:
I filed https://github.com/denoland/deno/issues/20013 to cover this -- we can remove this try/catch once we fix that issue.

Reply:
For now, let's add the simple catch and a note with the following:

```
// TODO(mmastrac): This should not terminate the server https://github.com/denoland/deno/issues/20013
```